### DISCUSSION TITLE
.NET6.0/WPF/VS2022 XAML-Designer: The name "ReactiveWindow" does not exist in the namespace "http://reactiveui.net".

### DISCUSSION THREAD

**Discussion by mimeister** [reactiveui/ReactiveUI#3457] [Q&A] [answered]
Hi there!

I started to re-write one of our WPF projects using ReactiveUI and have some trouble to get the XAML-Designer to accept the markup. After double-checking everything and spending hours on googling the error messages I finally created a fresh and clean bare minimum VisualStudio solution and the errors are still present there.
So for the sake of simplicity I will refer to that code in the following, which I uploaded here: https://github.com/mimeister/SampleReactiveProject. It is just what you get when creating a new .NET6 WPF project in VisualStudio 2022 with
 - a custom default root namespace
 - the ReactieUI.WPF nuget package installed
 - `MainWindow` moved into a sub-namespace `View` and converted to a `ReactiveWindow`
 - an empty viewmodel `MainWindowViewModel `created in a sub-namespace `ViewModel`.

## The main problem I'm facing
The following screenshot from VS should speak for itself:
![grafik](https://user-images.githubusercontent.com/14336987/210548101-66ffb2d8-8945-468d-8c2f-27216d662bb5.png)

However, the code generated from the XAML in the file `MainWindow.g.i.cs` actually looks good to me:
``` csharp
using ReactiveUI;
using SampleRoot.SampleReactiveProject.View;
using SampleRoot.SampleReactiveProject.ViewModel;
[...]

namespace SampleRoot.SampleReactiveProject.View {
    
    /// <summary>
    /// MainWindow
    /// </summary>
    public partial class MainWindow : ReactiveUI.ReactiveWindow<SampleRoot.SampleReactiveProject.ViewModel.MainWindowViewModel>, System.Windows.Markup.IComponentConnector {
        [...]
    }
} 
```

The `MainWindow` codebehind:
``` csharp
using ReactiveUI;
using SampleRoot.SampleReactiveProject.ViewModel;

namespace SampleRoot.SampleReactiveProject.View;

/// <summary>
/// Interaction logic for MainWindow.xaml
/// </summary>
public partial class MainWindow : ReactiveWindow<MainWindowViewModel>
{
    public MainWindow()
    {
        InitializeComponent();
    }
}
```

The viewmodel class:
``` csharp
using ReactiveUI;

namespace SampleRoot.SampleReactiveProject.ViewModel;

public class MainWindowViewModel : ReactiveObject
{
}
```

## What I did so far
- Double- (or rather tenfold-) checked that
    - the view- and viewmodel-classes definitely are in the namespaces defined in the XAML,
    - all other neccessary XML-namespaces are listed in the XAML, including the reactiveui.net-namespace,
    - there are no typos whatsoever
    - `x:TypeArguments` is present and correct. In fact, if I change `x:TypeArguments` to something else, I get an expected error, that the typearguments of `MainWindow`'s partial class declarations must match, which disappears again, when I enter the correct type again.
- Messed around with the default root namespace, as the actual project I'm working on defines one like `XXX.Projectname`. Of course I always changed every reference to the namespaces, so everything should be correct. Whatever I did to the root namespace, it didn't affect the errors in the XAML designer.
- Searched Google for other people posting similar issues. The best two results I could find are issues posted here in this repo:
    - #1684 seems to boil down to a missing XML-namespace and a bad `x:Class` declaration, which I checked both in my code, as written above.
    - #2742 looks almost identical to my problem, but the original author @Ehrode states, that moving the viewmodel into a separate folder and namespace and adding "a reference to System.Xaml" fixed it. I already did the former - although I can't image any plausible explanation, how this could fix the problem - and don't quite understand what exactly they mean by the latter.
 - Compared my code over and over again to another project inside the same solution I'm actually working on. That one was migrated to ReactiveUI by someone else some time ago, while it was still a .NET Framework 4.7 project. I've successfully migrated it to .NET6 including the conversion to the SDK project format. In that project the XAML designer is working just fine with `ReactiveWindow ` and `ReactiveUserControl`. Though, I can't find any significant differences in the basic setup of that project and its XAML files compared to my target project and the bare minimum project above.
 - Compared my code over and over again to the code in [A Compelling Example](https://www.reactiveui.net/docs/getting-started/compelling-example#a-compelling-example) and read a couple of other resources at reactiveui.net, trying to find any hints on what I might be missing out.
 - Messed around with traditional C# namespace declaration using curly braces and the newer file-scoped namespace declaration. Doesn't change a thing here, just as expected.

At this point I'm running out of ideas, what could possibly cause/fix my problem and just hope that someone here with more experience can give me some pointers how to get this working.

Thanks in advance,
Michael

**Comment by ChrisPulman** [comment, marked as answer]
@mimeister please see [Updated Code](https://github.com/mimeister/SampleReactiveProject/pull/1)
I have updated the project to remove the unnecessary code and added a binding to demonstrate
This is an issue with any solution using any overridden base class in WPF such as ReactiveWindow or ReactiveControl.
NOTE: see that the compelling Example also has the base class declarations removed in code relying on the XAML implementation.

I hope that this helps.

**Reply by mimeister** [reply]
Thanks for your quick answer, Chris!

I've just quickly cloned your code and opened it in VS. First it produced the same error in the designer, **BUT actually building** the project fixed it! So I went back to my unmodified sample code and did the same thing: it builds and suddenly the designer accepts the markup and shows the empty window.
Well, that was an easy fix, kinda embarassing 😅. I was so busy trying to find anything that might be wrong with the code to resolve the error message, that I didn't even think about just clicking "Build" anyway.
So I guess this is just some kind of bug or unexpected behavior in VS/the XAML designer.

However, thanks for pointing out, that declaring the base class in the code-behind isn't necessary, if already defined in XAML. I wasn't aware of that.

PS: I merged the PR so you don't need to keep the forked repo. If anyone else is interested in what changed or what the original code was, just look at Chris's commit or the initial commit respectively.
